File: pset6/caesar/caesar.py
import sys

ALPHABET_SIZE = 26

# Check proper usage
if len(sys.argv) != 2:
    print('Usage: python caesar.py k, where k is a non-negative number for "key"')
    sys.exit(1)

# Get the key
try:
    key = int(sys.argv[1])
except ValueError:
    key = 0

# Get the plain text from user
text = input("plaintext:  ")
print("ciphertext: ", end="")

# Calculate the cipher text and print it out char by char
for c in text:
    if c.isalpha():
        if c.islower():
            c = chr((ord(c) - ord('a') + key) % ALPHABET_SIZE + ord('a'))
        else:
            c = chr((ord(c) - ord('A') + key) % ALPHABET_SIZE + ord('A'))
    print(c, end="")

# Print a newline
print()

File: pset2/caesar/caesar.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define ALPHABET_SIZE 26

int main(int argc, string argv[])
{
    if (argc != 2) // finish program with error if it was started in an unappropriate way
    {
        printf("Usage: ./caesar k, where k is a non-negative number for \"key\"\n");
        return 1;
    }

    int k = atoi(argv[1]); // convert the key from string to int using atoi function from stdlib.h
    string p = get_string("plaintext:  "); // get the plain text from user

    for (int i = 0, n = strlen(p); i < n; i++) // iterate the plain text; encipher char by char
    {
        if (isalpha(p[i])) // we need to encipher only the letters
        {
            if (islower(p[i])) // formula for lower case letters
            {
                // p[i] - 'a' converts the "ASCII order" of a character to the alphabetical order, appliable in the formula
                // p[i] += 'a' converts it back to ASCII order
                // the same applies down for upper case letters
                // we save the cipher text over the same variable p since there is no need for the plain text in the future
                p[i] = (p[i] - 'a' + k) % ALPHABET_SIZE;
                p[i] += 'a';
            }
            else if (isupper(p[i])) // formula for upper case letters
            {
                p[i] = (p[i] - 'A' + k) % ALPHABET_SIZE;
                p[i] += 'A';
            }
        }

    }

    printf("ciphertext: %s\n", p); // print the cipher text
    return 0;
}